lib/utils.js

```diff
@@ -25,16 +25,26 @@ function isMergeableObject(val) {
 /**
 * @param {Object} target
 * @param {Object} source
+* @param {Boolean} shouldRemoveNullObjectValues
 * @returns {Object}
 */
-function mergeObject(target, source) {
+function mergeObject(target, source, shouldRemoveNullObjectValues = true) {
+    const targetAndSourceIdentical = target === source;
+
     const destination = {};
     if (isMergeableObject(target)) {
         // lodash adds a small overhead so we don't use it here
         // eslint-disable-next-line rulesdir/prefer-underscore-method
         const targetKeys = Object.keys(target);
         for (let i = 0; i < targetKeys.length; ++i) {
             const key = targetKeys[i];
+
+            // If shouldRemoveNullObjectValues is true, we want to remove null values from the merged object
+            if (shouldRemoveNullObjectValues && (_.isNull(target[key]) || _.isNull(source[key]))) {
```

Comment:
Can we remove lodash from this function? Lodash is slow and this function is used everywhere.